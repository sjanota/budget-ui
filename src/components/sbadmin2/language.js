import React, { createContext, useContext } from 'react';
import { useSBAdmin2 } from './context';
import PropTypes from 'prop-types';
import { capitalize } from '../../util/capitalize';

const DictionaryContext = createContext();

export function DictionaryProvider({ dictionaries, children }) {
  const { user } = useSBAdmin2();
  return (
    <DictionaryContext.Provider value={dictionaries[user.locale]}>
      {children}
    </DictionaryContext.Provider>
  );
}

DictionaryProvider.propTypes = {
  dictionaries: PropTypes.object,
  children: PropTypes.node,
};

export const useDictionary = () => useContext(DictionaryContext);

export function withDictionary(prop, Component, readPropName) {
  return props => {
    readPropName = readPropName || `read${capitalize(prop)}`;
    const readDict = props[readPropName];
    const newProps = { ...props };
    delete newProps[readPropName];
    const dictionary = useDictionary();
    if (readDict) {
      newProps[prop] = readDict(dictionary);
    }
    return <Component {...newProps} />;
  };
}
