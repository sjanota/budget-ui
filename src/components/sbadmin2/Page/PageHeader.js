import React from 'react';
import PropTypes from 'prop-types';
import { withDictionary } from '../language';

function PageHeader({ title }) {
  return <h1 className="h3 mb-4 text-gray-800">{title}</h1>;
}

PageHeader.propTypes = {
  title: PropTypes.string.isRequired,
};

export default withDictionary('title', PageHeader);
