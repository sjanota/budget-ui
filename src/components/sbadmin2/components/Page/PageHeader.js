import PropTypes from 'prop-types';
import React from 'react';

import { withDictionary } from '../../language';

function PageHeader({ children, title }) {
  return (
    <h1 className='h3 mb-4 text-gray-800 d-flex justify-content-between'>
      {title}
      {children && <div>{children}</div>}
    </h1>
  );
}

PageHeader.propTypes = {
  title: PropTypes.string.isRequired,
};

export default withDictionary('title', PageHeader);
