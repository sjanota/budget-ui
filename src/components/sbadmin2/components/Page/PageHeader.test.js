import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import PageHeader from './PageHeader';
import { DictionaryContext } from '../../language';

describe('PageHeader', () => {
  it('uses provided with title', () => {
    const component = mount(<PageHeader title="My page" />);
    expect(component).toHaveText('My page');
  });

  it('reads title from dictionary', () => {
    const component = mount(
      <DictionaryContext.Provider value={{ title: 'My page' }}>
        <PageHeader readTitle={d => d.title} />
      </DictionaryContext.Provider>
    );
    expect(component).toHaveText('My page');
  });

  it('renders with minimal props', () => {
    const component = renderer.create(<PageHeader title="" />);
    expect(component).toMatchSnapshot();
  });
});
