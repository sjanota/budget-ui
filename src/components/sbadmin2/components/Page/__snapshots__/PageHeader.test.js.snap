// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader renders with minimal props 1`] = `
<h1
  className="h3 mb-4 text-gray-800"
>
  
</h1>
`;
